metacatalog/util/results.py

```diff
@@ -396,7 +396,7 @@ def get_data(self, verbose=False, merge=False, **kwargs) -> dict:
             # merge everything that is a DataFrame
             for checksum, dat in data.items():
                 if isinstance(dat, pd.DataFrame):
-                    all_data = pd.merge(all_data, dat, left_index=True, right_index=True, how='outer')
+                    all_data = pd.concat([all_data, dat])
```

Comment:
I think this only works for Split datasets as Composites were handled correctly. For a composite, the data resulting from different entries should be *merged* along a common index axis